src/containers/Transaction/Transaction.js

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from 'react';
+import Switch from '@material-ui/core/Switch';
+import TextField from '@material-ui/core/TextField';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import Button from '@material-ui/core/Button';
+import { wrapper, heading, form, buttonContainer } from './transaction.module.scss';
+import Text from '../../components/Text';
+import useStockExchangeData from '../../hooks/useStockExchangeData';
+
+const Transaction = () => {
+  const [checked, setType] = useState(true);
+  const [currency, setCurrency] = useState('');
+  const [price, setPrice] = useState('');
+  const [amount, setAmount] = useState('');
+  const [total, setTotal] = useState('');
+
+  const [amountProps, setAmountProps] = useState({ error: false, helperText: '' });
+  const [currenciesProps, setCurrenciesProps] = useState({ error: false });
+
+  const [currencies, setCurrencies] = useState([]);
```

Comment:
A lot of states for pretty small form huh? Thats why we use libs like Formik etc. However we can keep that in that "raw" form with little improvement... How about to prepare custom hook for handle that form? https://www.telerik.com/blogs/how-to-build-custom-forms-react-hooks